### ISSUE TITLE
Finalize integration of battles with npc team

### ISSUE THREAD

**Issue by Elsa-Athiley** [uchicago-cs/chiventure#936]
Last sprint we met with the npc team to talk about integrating npc's into battles we will continue to think about ways to do this and begin implementing our ideas. The first step is to move the enemy_struct currently in battle code to npc code however ,issue #935 has to be finished first because the duplication of structs is also causing problems for the npc team.

**Comment by Elsa-Athiley** [comment]
This issue has been drawn out to sprint four. We were not able to make as much progess as we wanted in the past sprint because the NPC team needed to wait on some of our other battle code PR's to go through that were causing errors in their code. All these problems should be solved now and we can continue with integration. Once their code is pushed we will use their implementation of an npc enemy rather than our own.

**Comment by Elsa-Athiley** [comment]
There has not been much work done on our end with npc battle integration. I am currently waiting for PR #899 to be merged. This PR by the npc team includes a new struct called npc_battle_t which holds information about npcs that is only pertinent to battles. Once this is merged we will edit our code to work with this struct instead of the current npc_enemy struct that we have.

**Comment by Elsa-Athiley** [comment]
The PR from the npc team has been merged and is now available, next steps include replacing the use of npc_enemy in our code to instead use the npc_battle field in the npc struct. Adjustments will also need to be made for new fields such as `hostility` and` surrender_level.`

**Comment by lczielinski** [comment]
We have successfully created and merged a pull request to integrate the npc_t struct with our code: #1104. We replaced the old npc_enemy_t struct with the npc_t struct (containing npc_battle_t, which has most of the fields used specifically for battle). We updated function definitions, tests, and examples to integrate with this new struct. Now, there can exist npcs without a battle struct; in this case, it would not be possible to battle with them. The npc_battle_t struct contains three fields that still need to be used/integrated into the battle code: int health, hostility_t hostility_level, and int surrender_level, which will become a backlog task.
